### PR TITLE
Change Table.columns keys(), values() to return generator

### LINKED ISSUE #10525: Should TableColumns keys and values be generators now?
I stumbled upon this:

https://github.com/astropy/astropy/blob/1f75aa164539949723971d5b767f9517a3d77344/astropy/table/table.py#L285-L290

The comment said they were defined for PY2/PY3 compatibility. We have dropped PY2 since v3.0. Ideally, we should also be able to remove these if they were indeed only for compatibility. However, they revert to PY2 behavior, which makes removal impossible in the PY3 world. Should `keys` and `values` really by generators now?

### PR DESCRIPTION
**Pull request by taldcroft**
### Description

This pull request is to make `Table.columns` (which is a dict subclass) return a generator for `keys()` and `values()` methods, instead of the current Py2-compatible `list`.

Fixes #10525

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -53,6 +53,10 @@ astropy.stats
 astropy.table
 ^^^^^^^^^^^^^
 
+- Change ``Table.columns.keys()`` and ``Table.columns.values()`` to both return
```
Comment (pllim) [inline CHANGES.rst:56]: Is the duplicate entry intentional?

**Comment by pllim** [comment]
Looks like you missed a doctest example:
```
095   # Make a new table with the same number of rows and add columns to original table
096   >>> t2 = Table(np.arange(25).reshape(5, 5), names=('e', 'f', 'g', 'h', 'i'))
097   >>> t.add_columns(t2.columns.values())
UNEXPECTED EXCEPTION: TypeError("'odict_values' object does not support indexing")
Traceback (most recent call last):
  File "/opt/python/3.7.1/lib/python3.7/doctest.py", line 1329, in __run
    compileflags, 1), test.globs)
  File "<doctest modify_table.rst[29]>", line 1, in <module>
  File "/home/travis/build/astropy/astropy/.tox/py37-test/lib/python3.7/site-packages/astropy/table/table.py", line 2086, in add_columns
    self.add_column(cols[ii], index=indexes[ii], name=names[ii],
TypeError: 'odict_values' object does not support indexing
/home/travis/build/astropy/astropy/docs/table/modify_table.rst:97: UnexpectedException
```

File: CHANGES.rst (hunk 1 above)
Comment (taldcroft) [reply CHANGES.rst:56]: Yes.  It's definitely an API change.  I put it in the new features just to give it more visibility, but I can see the point for taking it out of new features. Either way is fine by me.

**Comment by taldcroft** [comment]
I fixed the doctest problem.

**Comment by pllim** [comment]
Technically, there is no new feature introduced, just API change, but I'll defer to @mhvk .

**Comment by mhvk** [comment]
Agreed with @pllim - just API change. Thanks!

**Pull request merged**